Runtime/BacktraceDatabase.cs

```diff
@@ -534,8 +534,9 @@ private bool ValidateDatabaseSize()
             //check how many records are stored in database
             //remove in case when we want to store one more than expected number
             //If record count == 0 then we ignore this condition
-            if (BacktraceDatabaseContext.Count() + 1 > DatabaseSettings.MaxRecordCount && DatabaseSettings.MaxRecordCount != 0 && !BacktraceDatabaseContext.RemoveLastRecord())
-            {
+            var noMoreSpaceForReport = BacktraceDatabaseContext.Count() + 1 > DatabaseSettings.MaxRecordCount && DatabaseSettings.MaxRecordCount != 0;
```

Comment:
Why this new semantic? Previously we used to remove the last record.

Reply:
We're removing this error later - if it's necessary. Previously with this behavior we will override an error with deduplication counter.